-1

Tested at: ced0e2583eb1b4db438faf599dca68c8ca1a22a6


The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
https://github.com/cms-sw/cmssw/commit/40f7ae93edeaf236d7a1362cdf835b8f5a283552
https://github.com/cms-sw/cmssw/commit/fa3bd535ec4b75fe267e014678311cdf5caef89e
https://github.com/cms-sw/cmssw/commit/2b45b69020766e6fc90334ae5d5e2207b61504f7
https://github.com/cms-sw/cmssw/commit/d542e8d5b1506c2f8bbe9b49c3abc4ca8a5a6b58
https://github.com/cms-sw/cmssw/commit/f1997166fe9d9f333035ed71be6b7cb65f81d041
https://github.com/cms-sw/cmssw/commit/2b512edd80a5e3e1c019b749696022cbe978b4d2
https://github.com/cms-sw/cmssw/commit/9ce916614df933d8d420e971181d66f7ac45ff7d
You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-20952/23841/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-20952/23841/git-merge-result

You can see the results of the tests here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-20952/23841/summary.html

I found follow errors while testing this PR

**Failed tests:  Build  ClangBuild**

* **Build**:

I found an error when building: 
 
 <pre>>> Compiling  /build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_4_X_2017-10-17-2300/src/Utilities/StaticAnalyzers/src/FunctionDumper.cpp 
>> Compiling  /build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_4_X_2017-10-17-2300/src/Utilities/StaticAnalyzers/src/ArgSizeChecker.cpp 
>> Compiling  /build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_4_X_2017-10-17-2300/src/Utilities/StaticAnalyzers/src/ConstCastChecker.cpp 
>> Compiling  /build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_4_X_2017-10-17-2300/src/Utilities/StaticAnalyzers/src/ConstCastAwayChecker.cpp 
/build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_4_X_2017-10-17-2300/src/GeneratorInterface/Core/plugins/GenXSecAnalyzer.cc: In member function 'virtual void GenXSecAnalyzer::endJob()':
/build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_4_X_2017-10-17-2300/src/GeneratorInterface/Core/plugins/GenXSecAnalyzer.cc:576:15: error: 'thisEventEffStat' was not declared in this scope
           << (thisEventEffStat.filterEfficiency(+3) * 100) << " +/- "
               ^~~~~~~~~~~~~~~~
>> Compiling  /build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_4_X_2017-10-17-2300/src/Utilities/StaticAnalyzers/src/edmChecker.cpp 
>> Compiling  /build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_4_X_2017-10-17-2300/src/Utilities/StaticAnalyzers/src/MutableMemberChecker.cpp 
>> Compiling  /build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_4_X_2017-10-17-2300/src/Utilities/StaticAnalyzers/src/ClassChecker.cpp 
</pre>

* **Clang**:

I found a compilation error while trying to compile with clang: 
I used this command: 
 scram b vclean && scram build -k -j 48 USER_CXXFLAGS='-fsyntax-only' COMPILER='llvm compile'


 <pre>>> Compiling edm plugin /build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_4_X_2017-10-17-2300/src/DQMOffline/Trigger/src/JetMETHLTOfflineSource.cc 
1 warning generated.
>> Compiling edm plugin /build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_4_X_2017-10-17-2300/src/DQMOffline/Trigger/src/HLTTauCertifier.cc 
>> Compiling edm plugin /build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_4_X_2017-10-17-2300/src/DQMOffline/Trigger/src/EgHLTOfflineSource.cc 
>> Compiling edm plugin /build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_4_X_2017-10-17-2300/src/DQMOffline/Trigger/src/EventShapeDQM.cc 
/build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_4_X_2017-10-17-2300/src/GeneratorInterface/Core/plugins/GenXSecAnalyzer.cc:576:15: error: use of undeclared identifier 'thisEventEffStat'
          << (thisEventEffStat.filterEfficiency(+3) * 100) << " +/- " 
              ^
>> Compiling edm plugin /build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_4_X_2017-10-17-2300/src/DQMOffline/Trigger/src/HotlineDQM.cc 
/build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_4_X_2017-10-17-2300/src/GeneratorInterface/Core/plugins/GenXSecAnalyzer.cc:577:16: error: use of undeclared identifier 'thisEventEffStat'
          << ( thisEventEffStat.filterEfficiencyError(+3) * 100) <<"   [TO BE USED IN MCM]";
</pre>


The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
https://github.com/cms-sw/cmssw/commit/40f7ae93edeaf236d7a1362cdf835b8f5a283552
https://github.com/cms-sw/cmssw/commit/fa3bd535ec4b75fe267e014678311cdf5caef89e
https://github.com/cms-sw/cmssw/commit/2b45b69020766e6fc90334ae5d5e2207b61504f7
https://github.com/cms-sw/cmssw/commit/d542e8d5b1506c2f8bbe9b49c3abc4ca8a5a6b58
https://github.com/cms-sw/cmssw/commit/f1997166fe9d9f333035ed71be6b7cb65f81d041
https://github.com/cms-sw/cmssw/commit/2b512edd80a5e3e1c019b749696022cbe978b4d2
https://github.com/cms-sw/cmssw/commit/9ce916614df933d8d420e971181d66f7ac45ff7d
You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-20952/23841/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-20952/23841/git-merge-result

--- a/GeneratorInterface/Core/plugins/GenXSecAnalyzer.cc
+++ b/GeneratorInterface/Core/plugins/GenXSecAnalyzer.cc
@@ -496,6 +496,8 @@ GenXSecAnalyzer::endJob() {
     unsigned int i = 0;
     double jetmatch_eff=0;
     double jetmatch_err=0;
+    double matching_eff=1;
+    double matching_efferr=1;
 
     for(std::map<int, GenFilterInfo>::const_iterator iter = jetMatchEffStat_.begin();
           iter!=jetMatchEffStat_.end(); ++iter, i++){ 
@@ -554,6 +556,8 @@ GenXSecAnalyzer::endJob() {
           << (thisEventEffStat.filterEfficiency(+3) * 100) << " +/- " 
           << ( thisEventEffStat.filterEfficiencyError(+3) * 100);
 
+      matching_eff = thisEventEffStat.filterEfficiency(+3);
+      matching_efferr = thisEventEffStat.filterEfficiencyError(+3);
     }
     delete [] title;
 
@@ -569,12 +573,14 @@ GenXSecAnalyzer::endJob() {
       << "After matching: total cross section = " 
       << std::scientific << std::setprecision(3)  
       << xsecAfterMatching_[last].value() << " +- " << xsecAfterMatching_[last].error() <<  " pb";
-      
+
+          
     edm::LogPrint("GenXSecAnalyzer")
       << "Matching efficiency = "
       << std::fixed << std::setprecision(1)
-          << (thisEventEffStat.filterEfficiency(+3) * 100) << " +/- " 
-          << ( thisEventEffStat.filterEfficiencyError(+3) * 100) <<"   [TO BE USED IN MCM]";
+      <<  matching_eff << " +/- " 
+      <<  matching_efferr <<"   [TO BE USED IN MCM]";
+    
   }
   else if(hepidwtup_ == -1 )
     edm::LogPrint("GenXSecAnalyzer") 

